apps/example-orchestrator-ui/components/Subscription/Subscription.tsx

```diff
@@ -0,0 +1,176 @@
+                    2,
+                ),
+        },
+    );
+
+    if (dataComplete && subscriptionData.subscriptionId === '') {
+        console.log('No data in context populating dataComplete from cache');
+        setSubscriptionData(
+            mapApiResponseToSubscriptionDetail(dataComplete, true),
+            2,
+        );
+    } else if (
+        !dataComplete &&
+        data &&
+        subscriptionData.subscriptionId === ''
+    ) {
+        console.log('No data in context populating dataOutline from cache');
+        setSubscriptionData(mapApiResponseToSubscriptionDetail(data, false), 1);
+    }
+
+    const renderTabs = () => {
+        return tabs.map((tab, index) => (
+            <EuiTab
+                key={index}
+                onClick={() => onSelectedTabChanged(tab.id)}
+                isSelected={tab.id === selectedTabId}
+                disabled={tab.disabled}
+                prepend={tab.prepend}
+                append={tab.append}
+            >
+                {tab.name}
+            </EuiTab>
+        ));
+    };
+
+    return (
+        <>
+            <EuiFlexGroup
+                style={{ marginBottom: 10 }}
+                justifyContent="spaceBetween"
+            >
+                <EuiFlexItem grow={true}>
+                    <EuiText>
+                        <h2>
+                            {isLoading ? '' : data?.subscription?.description}
+                        </h2>
+                    </EuiText>
+                </EuiFlexItem>
+                <EuiFlexItem grow={false}>
+                    <EuiFlexGroup justifyContent="spaceBetween">
+                        <EuiFlexItem style={{ width: 140 }}>
+                            <span style={{ marginTop: 5 }}>
+                                Loading status
+                                <EuiBadge
+                                    style={{ marginLeft: 4 }}
+                                    color={getColor(loadingStatus)}
+                                >
+                                    {loadingStatus}
+                                </EuiBadge>
+                            </span>
+                        </EuiFlexItem>
+                        <EuiFlexItem>
+                            <SubscriptionActions
+                                subscriptionId={subscriptionId}
+                            />
+                        </EuiFlexItem>
+                    </EuiFlexGroup>
+                </EuiFlexItem>
+            </EuiFlexGroup>
+            <>
+                <EuiTabs>{renderTabs()}</EuiTabs>
+                {selectedTabContent}
+            </>
+
+            {selectedTabId === GENERAL_TAB && <SubscriptionGeneral />}
+            {selectedTabId === SERVICE_CONFIGURATION_TAB && (
+                <SubscriptionDetailTree />
+            )}
+            {selectedTabId === PROCESSES_TAB && data && (
+                <ProcessesTimeline subscriptionId={subscriptionId} />
+            )}
+        </>
+    );
+};
```

Comment:
It has probably been mentioned before in the gui meetings and I failed to remember, but is there anything keeping us from moving this component into the library? Is it because of the query/mapResponse functions which are implementation specific and need to be imported in this component?

Reply:
Yes. This component is responsible for:
- fetching the data and mapping it to `SubscriptionDetailBase` : so it contains GPL / Surf specific logic
- I assume that some customers want to add extra tabs etc: this showcases how that can be done.